Support text-2.0

`cabal build --constraint 'text >= 2.0' --allow-newer='monoid-subclasses:text'` fails with 
```
src/Data/Monoid/GCD.hs:39:36: error:
    Module ‘Data.Text.Unsafe’ does not export ‘lengthWord16’
   |
39 | import           Data.Text.Unsafe (lengthWord16, reverseIter)
   |
```

--- a/monoid-subclasses.cabal
+++ b/monoid-subclasses.cabal
@@ -30,7 +30,7 @@ Library
                      Data.Monoid.Instances.Concat, Data.Monoid.Instances.Measured, Data.Monoid.Instances.Positioned,
                      Data.Monoid.Instances.Stateful
   Build-Depends:     base >= 4.9 && < 5,
-                     bytestring >= 0.9 && < 1.0, containers >= 0.5.7.0 && < 0.7, text >= 0.11 && < 1.3,
+                     bytestring >= 0.9 && < 1.0, containers >= 0.5.7.0 && < 0.7, text >= 0.11 && < 1.3 || >= 2.0 && < 2.1,
                      primes == 0.2.*, vector >= 0.12 && < 0.13
   GHC-options:       -Wall
   default-language:  Haskell2010
@@ -38,7 +38,7 @@ Library
 test-suite Main
   Type:              exitcode-stdio-1.0
   Build-Depends:     base >= 4.9 && < 5,
-                     bytestring >= 0.9 && < 1.0, containers >= 0.5.7.0 && < 0.7, text >= 0.11 && < 1.3,
+                     bytestring >= 0.9 && < 1.0, containers >= 0.5.7.0 && < 0.7, text >= 0.11 && < 1.3 || >= 2.0 && < 2.1,
                      vector >= 0.12 && < 0.13, primes == 0.2.*,
                      QuickCheck >= 2.9 && < 3, quickcheck-instances >= 0.3.12 && <0.4,
                      tasty >= 0.7, tasty-quickcheck >= 0.7 && < 1.0,

--- a/src/Data/Monoid/GCD.hs
+++ b/src/Data/Monoid/GCD.hs
@@ -36,7 +36,10 @@ import qualified Data.Text as Text
 import qualified Data.Text.Encoding as TextEncoding
 import qualified Data.Text.Internal as Internal
 import qualified Data.Text.Internal.Lazy as LazyInternal
-import           Data.Text.Unsafe (lengthWord16, reverseIter)
+import           Data.Text.Unsafe (reverseIter)
+#if MIN_VERSION_text(2,0,0)
+import           Data.Text.Unsafe (Iter(..))
+#endif
 import qualified Data.Text.Lazy as LazyText
 import qualified Data.Text.Lazy.Encoding as LazyEncoding
 import qualified Data.IntMap as IntMap
@@ -393,8 +396,13 @@ instance RightGCDMonoid Text.Text where
                    | otherwise = (Internal.text xarr xoff (succ i),
                                   Internal.text yarr yoff (succ j),
                                   Internal.text xarr (xoff+i+1) (xlen-i-1))
+#if MIN_VERSION_text(2,0,0)
+               where Iter xc xd = reverseIter x i
+                     Iter yc yd = reverseIter y j
+#else
                where (xc, xd) = reverseIter x i
                      (yc, yd) = reverseIter y j
+#endif
 #else
   stripCommonSuffix x y =
     let (xlist, ylist, slist) =
@@ -421,7 +429,7 @@ instance RightGCDMonoid LazyText.Text where
             x0len = lazyLengthWord16 x0
             y0len = lazyLengthWord16 y0
             lazyLengthWord16 = LazyText.foldlChunks addLength 0
-            addLength n x = n + lengthWord16 x
+            addLength n x = n + (\(Internal.Text _ _ l) -> l) x
             splitWord16 xp 0 x = (xp, x)
             splitWord16 xp n (LazyInternal.Chunk x@(Internal.Text arr off len) xs)
                | n < len = (xp . LazyInternal.chunk (Internal.Text arr off n),

